File: apps/api/app/schemas/audit_log.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: uuid.UUID
    actor_id: uuid.UUID | None
    actor_email: str
    action: str
    entity_type: str
    entity_id: uuid.UUID | None
    entity_asset_id: str | None
    before_state: Any | None
    after_state: Any | None
    ip_address: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/web/src/types/audit_log.ts
export interface AuditLogEntry {
  id: string;
  actor_id: string | null;
  actor_email: string;
  actor_name: string | null;
  actor_role: string | null;
  action: string;
  entity_type: string;
  entity_id: string | null;
  entity_asset_id: string | null;
  before_state: unknown | null;
  after_state: unknown | null;
  ip_address: string | null;
  created_at: string;
}
